### PR TITLE
ADD: Task 13, 14 Test: 13,14

### PR DESCRIPTION
**Pull request by wenprompt**
## Summary by CodeRabbit

- New Features
  - Unified reconciliation result consolidating matches, position comparisons, summaries, and per-exchange positions.
  - Service layer enabling both async (API) and sync (CLI) reconciliation workflows.
  - CLI enhancements: optional JSON output, improved input validation, clearer errors, and explicit exit codes.

- Documentation
  - Updated product docs to reflect the consolidated reconciliation flow and CLI behavior.

- Tests
  - Extensive unit tests for consolidation logic, service workflows (sync/async), and routing validation, covering realistic multi-exchange scenarios and edge cases.

### CODE REVIEW

File: src/reconciliation/consolidator.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        all_exchange_pos: dict[str, list[BaseTrade]] = {}
+        summary: dict[str, dict[str, Any]] = {}
+
+        # Merge each exchange result
+        for result in exchange_results:
+            # Combine matches into single list
+            all_matches.extend(result.matches)
+
+            # Keep position comparisons separated by exchange
+            all_position_comparisons[result.exchange_name] = result.position_comparisons
```
[the remark this answers is not in the document]
Comment (wenprompt) [reply src/reconciliation/consolidator.py:83]: architecturely there can only be 1 instance of each exchange group

File: tests/fixtures/exchange_results.py [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+    }
+
+    return ExchangeResult(
+        exchange_name="SGX",
+        matches=matches,
+        position_comparisons=position_comparisons,
+        statistics=statistics,
+        trader_pos=trader_pos,
+        exchange_pos=exchange_pos,
+    )
```
[the remark this answers is not in the document]
Comment (wenprompt) [reply tests/fixtures/exchange_results.py:216]: modified and statistics removed from all implementations

File: docs/tasks.md [hunk 3]
```diff
@@ -447,63 +447,63 @@ Build complete framework skeleton with hexagonal architecture, frozen dataclasse
...
+    - [x] Call `self.consolidator.merge(exchange_results)`
+    - [x] Return consolidated result
+    - [x] Add docstring: "Single method for both API and CLI"
+  - [x] `process_sync(json_path: Path) -> ConsolidatedResult`:
+    - [x] Call load_json_file to get trader_data, exchange_data
+    - [x] Build json_data dict with "traderTrades" and "exchangeTrades"
+    - [x] Call `self.router.route_and_process(json_data)`
+    - [x] Return `self.consolidator.merge(exchange_results)`
+    - [x] Add docstring: "Synchronous version for CLI"
+- [x] Add class docstring
```
[the remark this answers is not in the document]
Comment (wenprompt) [reply docs/tasks.md:506]: error propagated, API not implemented in this phase
